files/build_templates/docker_image_ctl.j2

```diff
@@ -227,6 +227,11 @@ start() {
 {%- else %}
         --tmpfs /tmp \
 {%- endif %}
+{%- endif %}
```

Comment:
The Sonic-buildimage, builds a debug version  when INSTALL_DEBUG_TOOLS=y. This debug version has full symbols loaded for each binary.  Refer this [commit](https://github.com/Azure/sonic-buildimage/commit/ba0ca01ee052e2c08f6390b65a0fa3aa4d0f5fdb).

We also archive the source into image.

In this [commit](https://github.com/Azure/sonic-buildimage/commit/76bf5a0bc4bde568e4ffc0e12cafe3f13520400b), we map the archived source path as ro and an empty dir /debug as rw for any writes from inside the container during debug session. For cores that is created by an image running non-debug/production version, users can copy into /debug and analyze by running gdb inside the container.

As you may note, all this is controlled by INSTALL_DEBUG_TOOLS.

In sync with Joe, we could rally all this behind this single flag.

Also you may need to see, how this work complements/in-harmony-with  the current debug strategy, so we can have a holistic approach to debugging in SONiC.

Another NOTE: Only in dev environment, we have the need to analyze core dropped in the box. Most common / frequent use case is, a production box drops core, which is collected and analyzed outside the switch (as we can't afford to do any debugging inside a switch running production traffic).

Reply:
Agreed and I made the change to use iNSTALL_DEBUG_TOOLS instead of SONIC_DEBUGGING_ON.